lunatrace/bsl/frontend/src/components/auth/RouteGuard.tsx

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright by LunaSec (owned by Refinery Labs, Inc)
+ *
+ * Licensed under the Business Source License v1.1
+ * (the "License"); you may not use this file except in compliance with the
+ * License. You may obtain a copy of the License at
+ *
+ * https://github.com/lunasec-io/lunasec/blob/master/licenses/BSL-LunaTrace.txt
+ *
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+import React, { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import useAppDispatch from '../../hooks/useAppDispatch';
+import { useAppSelector } from '../../hooks/useAppSelector';
+import { clear } from '../../store/slices/alerts';
+import { selectConfirmedUnauthenticated } from '../../store/slices/authentication';
+
+export const RouteGuard: React.FC = (props) => {
+  console.log('route guard rendering');
+  const confirmedNotAuthenticed = useAppSelector(selectConfirmedUnauthenticated);
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  useEffect(() => {
+    console.log('isinitialized is ', confirmedNotAuthenticed);
+    if (confirmedNotAuthenticed) {
+      console.warn('Detected use not logged in, redirecting to login page and saving URL for redirect back afterwards');
+
+      // navigate to the login screen, saving the current url as a redirect that kratos will pick up after login
+      navigate(`/?return_to=${window.location.href}`);
```

Comment:
Security Nit: You should do `encodeUriComponent(window.location.href)` to prevent any potential path pollution attacks.